src/platform_impl/linux/wayland/seat/dnd/handlers.rs

```diff
@@ -0,0 +1,185 @@
+use std::io::{self, Read};
+use std::os::unix::prelude::{AsRawFd, RawFd};
+use std::path::PathBuf;
+use std::str;
+
+use percent_encoding::percent_decode_str;
+use sctk::data_device::{DataOffer, DndEvent, ReadPipe};
+use sctk::reexports::calloop::{generic::Generic, Interest, LoopHandle, Mode, PostAction};
+
+use crate::dpi::PhysicalPosition;
+use crate::event::WindowEvent;
+use crate::platform_impl::wayland::{event_loop::WinitState, make_wid, DeviceId};
+
+use super::DndInner;
+
+const MIME_TYPE: &str = "text/uri-list";
+
+pub(super) fn handle_dnd(event: DndEvent<'_>, inner: &mut DndInner, winit_state: &mut WinitState) {
+    match event {
+        DndEvent::Enter {
+            offer: Some(offer),
+            surface,
+            x,
+            y,
+            ..
+        } => {
+            let window_id = make_wid(&surface);
+            inner.window_id = Some(window_id);
+            offer.accept(Some(MIME_TYPE.into()));
+            let _ = parse_offer(&inner.loop_handle, offer, move |paths, winit_state| {
```

Comment:
None of these is needed.

Reply:
The callback emits the `HoveredFile` event though?

Reply:
Hm, yeah. Sure. I'm not sure why anyone would need it. but ok.